File: request-manager/src/main/java/fii/request/manager/domain/Customer.java
package fii.request.manager.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Entity
@Data
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long customerId;
    private String firstName;
    private String lastName;
    private Role role;

    @ManyToMany(cascade = CascadeType.ALL)
    @JoinTable(
            name = "Customer_To_Group",
            joinColumns = { @JoinColumn(name = "customer_id") },
            inverseJoinColumns = { @JoinColumn(name = "group_id") }
    )
    private Set<CustomerGroup> customerGroups = new HashSet<>();
}

File: workflow-manager-frontend/app/src/main/java/com/example/workflow_manager_frontend/domain/Customer.kt
package com.example.workflow_manager_frontend.domain

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Customer(
    val email: String,
    val firstName: String?,
    val id: Int,
    val lastName: String?,
    val role: String?,
    val password: String?
) : Parcelable {
    override fun toString(): String {
        return email
    }
}